### PR TITLE
Feature etb#95 adding gdb runner

### PR DESCRIPTION
**Pull request by jrachucki-n7s**

### CODE REVIEW

File: Config/armbsp.cfg [hunk 1]
```diff
@@ -0,0 +1,33 @@
+[gdbServer]
+address = %(ARMBSP_RASPI_ADDRESS)s
```
Comment (Lurkerpas) [inline Config/armbsp.cfg:2]: Why ARMBSP?

File: Config/armbsp.cfg [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+baudrate = 115200
+port = 5005
+path = /dev/ttyACM0
+verbose = True
+
+[ioUart4]
+address = %(ARMBSP_RASPI_ADDRESS)s
+userName = %(TASTE_RASPI_USER)s
+password = %(TASTE_RASPI_PASSWORD)s
+baudrate = 38400
```
Comment (Lurkerpas) [inline Config/armbsp.cfg]: I would recommend to make it lower. Or, reconfigurable with a lower default (stable 9600 for tests, potentially higher for demos/etc)

File: Config/armbsp.cfg (hunk 1 above)
Comment (Lurkerpas) [reply Config/armbsp.cfg:2]: Oh... I see. This is the second config.

File: Config/taste.cfg [hunk 3]
```diff
@@ -0,0 +1,35 @@
...
+baudrate = 115200
+port = 5005
+path = /dev/ttyACM0
+verbose = True
+
+[ioUart4]
+address = %(TASTE_RASPI_ADDRESS)s
+userName = %(TASTE_RASPI_USER)s
+password = %(TASTE_RASPI_PASSWORD)s
+baudrate = 38400
```
Comment (Lurkerpas) [inline Config/taste.cfg]: I would recommend to make it lower. Or, reconfigurable with a lower default (stable 9600 for tests, potentially higher for demos/etc)

File: Config/armbsp.cfg (hunk 1 above)
Comment (jrachucki-n7s) [reply Config/armbsp.cfg:2]: I was using this configuration to connect to armbsp setup when taste-setup was not available.

File: gdb_runner.py [hunk 4]
```diff
@@ -0,0 +1,224 @@
...
+                config["baudrate"],
+                config["port"],
+                config["vPortName"],
+                debug=config["verbose"],
+            )
+        handler.open()
+        return handler
+
+
+    def __dumpLogLambda(self, io, logPath):
```
Comment (Lurkerpas) [inline gdb_runner.py]: What is lambda?

Comment (Lurkerpas) [reply gdb_runner.py]: OK... I see... but I don't think this is really a lambda. I would rename it to e.g __dumpLogInternal or better __receiveAndWriteLog.

File: Config/armbsp.cfg [hunk 5]
```diff
@@ -0,0 +1,33 @@
+[gdbServer]
+address = %(ARMBSP_RASPI_ADDRESS)s
+userName = %(TASTE_RASPI_USER)s
```
Comment (Lurkerpas) [inline Config/armbsp.cfg]: Please harmonize the variable names

File: libs/GdbInterface.py [hunk 6]
```diff
@@ -0,0 +1,210 @@
...
+    def __init__(self, address, path="gdb", verbose=False):
+        self.address = address
+        self.verbose = verbose
+        self.path = path
+
+        self.gdbmi = None
+        self.running = False
+        self.launched = False
+
+    def printv(self, *args, **kwargs):
```
Comment (Lurkerpas) [inline libs/GdbInterface.py]: printIfVerbose? printConditional? one letter suffix clashes with printr

File: libs/GdbInterface.py [hunk 7]
```diff
@@ -0,0 +1,210 @@
...
+        if self.verbose:
+            msg = args[0].replace("\\n", "\n")
+            msg = msg.replace("\\t", "\t")
+            print(msg, *args[1:], **kwargs)
+
+    def printr(self, response):
```
Comment (Lurkerpas) [inline libs/GdbInterface.py]: printFormattedResponse? printResponse?

File: Config/armbsp.cfg [hunk 8]
```diff
@@ -0,0 +1,33 @@
...
+verbose = True
+
+[ioUart4]
+address = %(ARMBSP_RASPI_ADDRESS)s
+userName = %(TASTE_RASPI_USER)s
+password = %(TASTE_RASPI_PASSWORD)s
+baudrate = 38400
+port = 5006
+path = /dev/serial0
+vPortName = vUart4
```
Comment (Lurkerpas) [inline Config/armbsp.cfg]: This applies not only here, but also across python code. Would it be possible to make better names? This is virtualPortName? virtualDeviceName? Maybe sync this renaming with path -> hardwarePortPath? hardwareDevicePath? What do you think?

Comment (jrachucki-n7s) [reply Config/armbsp.cfg]: I agree. I will try to fix it across the whole project.

**Review by Lurkerpas — APPROVED** [review]

**Pull request merged**